src/java.base/share/classes/jdk/internal/misc/Blocker.java

```diff
@@ -62,64 +60,36 @@ private static Thread currentCarrierThread() {
     }
 
     /**
-     * Marks the beginning of a possibly blocking operation.
-     * @return the return value from the attempt to compensate or -1 if not attempted
+     * Marks the beginning of a blocking operation.
+     * @return true if tryCompensate attempted
      */
-    public static long begin() {
+    public static boolean begin() {
         if (VM.isBooted()
-                && currentCarrierThread() instanceof CarrierThread ct && !ct.inBlocking()) {
+                && Thread.currentThread().isVirtual()
+                && currentCarrierThread() instanceof CarrierThread ct) {
             ct.beginBlocking();
-            boolean completed = false;
-            try {
-                long comp = ForkJoinPools.beginCompensatedBlock(ct.getPool());
-                assert currentCarrierThread() == ct;
-                completed = true;
-                return comp;
-            } finally {
-                if (!completed) {
-                    ct.endBlocking();
-                }
-            }
+            return true;
         }
-        return -1;
+        return false;
     }
 
     /**
      * Marks the beginning of a possibly blocking operation.
      * @param blocking true if the operation may block, otherwise false
-     * @return the return value from the attempt to compensate, -1 if not attempted
-     * or blocking is false
+     * @return true if tryCompensate attempted
```

Comment:
Was the use the word `tryCompensate` intentional here? I don't see that method in this class or in the implementation of `CarrierThread.beginBlocking()`

Reply:
> Was the use the word `tryCompensate` intentional here? I don't see that method in this class or in the implementation of `CarrierThread.beginBlocking()`

It was named tryCompensate at one point so a left over from a long time ago.